pkg/service/checkpointz/checkpointz.go

```diff
@@ -45,6 +48,41 @@ func (h *Handler) V1Status(ctx context.Context, req *StatusRequest) (*StatusResp
 	return response, nil
 }
 
+// Slot returns the beacon slot for checkpointz.
+func (h *Handler) V1BeaconSlots(ctx context.Context, req *BeaconSlotsRequest) (*BeaconSlotsResponse, error) {
+	response := &BeaconSlotsResponse{}
+
+	slot, err := h.provider.GetCurrentSlot(ctx)
+	if err != nil {
+		return nil, err
+	}
+	response.Slots = []BeaconSlot{}
+	for i, val := uint64(slot), uint64(slot)-100; i > val && i >= 0; i -= 1 {
```

Comment:
@Savid I think we want this to decrement 32 slots at a time, and for the current slot to actually be `current finalized epoch / 32` to derive the slot on the epoch boundary